tools/torch/main.lua

```diff
@@ -485,11 +485,13 @@ if opt.optimState ~= '' then
 end
 
 local function updateConfusion(y,yt)
-    if confusion ~= nil then
-        confusion:batchAdd(y,yt)
+    if trainConfusion ~= nil then
+        trainConfusion:batchAdd(y,yt)
     end
 end
 
+local label_function = network.labelHook or function (input, dblabel) return dblabel end
```

Comment:
sorry for being picky but the naming convention is slightly inconsistent with the other changes (where camel case is being used for globals)